functions/Get-DbaSqlManagementObject.ps1

```diff
@@ -0,0 +1,88 @@
+﻿function Get-DbaSqlManagementObject
+{
+<#
+	.SYNOPSIS
+	Gets SQL Mangaement Object versions installed on the machine.
+
+	.DESCRIPTION
+	 The Get-DbaSqlManagementObject returns an object with the Version and the Add-Type Load Template for each version on the server.
+
+	.PARAMETER VersionNumber
+	This is the specific version number you are looking for. The function will look 
+	for that version only.
+	
+	.PARAMETER Silent
+	Use this switch to disable any kind of verbose messages
+	
+	.NOTES
+	Original Author: Ben Miller (@DBAduck - http://dbaduck.com)
+
+	Tags: SMO
+
+	Website: https://dbatools.io
+	Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+	License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+	.LINK
+	https://dbatools.io/Get-DbaSqlManagementObject
+
+	.EXAMPLE
+	Get-DbaSqlManagementObject
+	Returns all versions of SMO on the computer
+	
+	.EXAMPLE
+	Get-DbaSqlManagementObject -VersionNumber 13
+	Returns just the version specified. If the version does not exist then it will return nothing.
+	
+
+#>
+	[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $false, Position=0)]
+		[ValidateSet(9,10,11,12,13,14)]
+		[int]$VersionNumber = 0,
```

Comment:
I don't know that I would do a validation set for this because it will mean we have to maintain this command each time a new version of SQL Server comes out.

I would simply validate it is 9 or greater in the begin block and then move on. 

@potatoqualitee or @PowerDBAKlaas thoughts?